### PR TITLE
Object3D.detach()

### PR DESCRIPTION
**Pull request by EthanHermsey**
Added a detach method to Object3D. 
The attach method is very handy and I was wondered why there is no method that does the opposite. 

I found Mugen87's solution here [https://discourse.threejs.org/t/threejs-detach-child-object-from-its-parent/19407](https://discourse.threejs.org/t/threejs-detach-child-object-from-its-parent/19407) and would like to see it added.

The new method is a copy of the remove method with one added line that decomposes the worldMatrix after removing the child. I am not sure if this is the best way, I chose this because the remove method does not confirm if the object is actually removed ( as a child of 'this' object ) unless you add an event to the object, but that seems unlogical in this case.

### CODE REVIEW

File: src/core/Object3D.d.ts [hunk 1]
```diff
@@ -293,6 +293,11 @@ export class Object3D extends EventDispatcher {
 	 */
 	attach( object: Object3D ): this;
 
+	/**
+	 * Removes object as a child of this, while maintaining the object's world transform.
+	 */
+	detach( object: Object3D ): this;
+
```
Comment (Mugen87) [inline src/core/Object3D.d.ts:300]: It seems your branch is based on an old commit. The TS type declaration files were removed quite a while ago.

I suggest you rebase your branch and force-push to fix the PR.

Comment (EthanHermsey) [reply src/core/Object3D.d.ts:300]: I will. Strange, I would swear I just forked this.

**Comment by WestLangley** [comment]
>The attach method is very handy and I was wondered why there is no method that does the opposite.

The reason is explained in #16528 -- you can just `attach()` it somewhere else.

What, exactly, is your use case?

**Comment by EthanHermsey** [comment]
Yup. Scene.attach(). I didn't realize Scene is an extension of Object3D.. 
I feel silly. I've been using Three for a while, never knew.. consider this pr unsubmitted.
